### ISSUE TITLE
Parallel test speed

### ISSUE THREAD

**Issue by jacg** [next-exp/IC#311]
Could you all please run

1. `time bash manage.sh run_tests`
2. `time bash manage.sh run_tests_par`
3. `time bash manage.sh run_tess_par N`, for different values of `N`.

and report the times you see, along with the number of processors on which ran the timings.

Here are my results:

1. `run_tests`: 50s
2. `run_tests_par 2`: 36s.
3. `run_tests_par 3`: 28s.
4. `run_tests_par 4`: 26s.
5. `run_tests_par 5`: 27s.
6. `run_tests_par 6`: 30s.
7. `run_tests_par 7`: 33s.
8. `run_tests_par`: 33s.

In other words, I'd like to draw your attention to the fact that it would probably be worth your while to take advantage of our parallel testing capabilities!

Your feedback might allow us to set some sensible defaults, or tune something somewhere.

**Comment by gonzaponte** [comment]
real or user?

**Comment by jacg** [comment]
We care about how long you have to wait until you see the full results of the test suite, so wall clock, however that happens to be labeled in your version of the shell. (In my case, `total`).

Also note that the variance in the timings in nontrivial. (I just got a 26s run on 8 CPUs, compared to the earlier 33s.)

**Comment by abotas** [comment]
@gonzaponte i think for is it is real
run_tests: 39s
run_tests_par 2: 22s
run_tests_par 3: 26s
run_tests_par 4: 28s
run_tests_par 5: 30s

**Comment by gonzaponte** [comment]
Results in Canfranc (12 CPUs):
```
                 mean +- rms
run_tests      : 56.0 +- 1.1 
run_tests_par  : 66.0 +- 1.0
run_tests_par 2: 50.8 +- 1.7
run_tests_par 3: 43.2 +- 1.0
run_tests_par 4: 43.1 +- 1.1
run_tests_par 5: 44.2 +- 1.2
run_tests_par 6: 46.5 +- 0.8
run_tests_par 7: 49.2 +- 1.1
run_tests_par 8: 54.7 +- 1.3
 ```

Although I think these results may be compromised by other people using the machines.
